### PR TITLE
issue-13589: adjust docs for custom-false-values for checkbox type

### PR DESCRIPTION
**Pull request by leberknecht**
refs #13589 on symfony/symfony repo

### CODE REVIEW

**Comment by javiereguiluz** [comment]
@vudaltsov we'd need your review about this proposal related to forms. Thanks a lot!

**Comment by vudaltsov** [comment]
@javiereguiluz , thanx for pinging me on this :)

I agree with the contents of this PR, but I would also dedicate a paragraph to the `false_values` option in the `Field Options` section.

**Review by vudaltsov — CHANGES_REQUESTED** [review]
Let's add a section under `Field Options`.

**Review by linaori — APPROVED** [review]

**Review by xabbuh — CHANGES_REQUESTED** [review]

File: reference/forms/types/checkbox.rst [hunk 1]
```diff
@@ -6,12 +6,16 @@ CheckboxType Field
...
+true, if the box is unchecked, the value will be set to false. Optionally
+you can specify an array of values that, if submitted, will be evaluated
+to "false" as well (this differs from what HTTP defines, but can be handy
+if you want to handle submitted values like "0" or "false").
 
 +-------------+------------------------------------------------------------------------+
 | Rendered as | ``input`` ``checkbox`` field                                           |
 +-------------+------------------------------------------------------------------------+
 | Options     | - `value`_                                                             |
+|             | - `false_values`_                                                      |
```
[the remark this answers is not in the document]
Comment (leberknecht) [reply reference/forms/types/checkbox.rst]: Doh! Indeed, sorry. Added.

File: reference/forms/types/options/false_values.rst.inc [hunk 2]
```diff
@@ -0,0 +1,6 @@
+false_values
+~~~~~
```
Comment (xabbuh) [inline reference/forms/types/options/false_values.rst.inc]: this line must contain as many `~` characters as the length of the headline

File: reference/forms/types/options/false_values.rst.inc [hunk 3]
```diff
@@ -0,0 +1,6 @@
+false_values
+~~~~~~~~~~~~
+
+**type**: ``array`` **default**: ``array(null)``
+
+An array of values to be interpreted as `false`.
```
Comment (xabbuh) [inline reference/forms/types/options/false_values.rst.inc]: we need to backticks here

File: reference/forms/types/checkbox.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (xabbuh) [reply reference/forms/types/checkbox.rst]: Oh and maybe we should sort the options alphabetically (i.e. put `false-values` before `value`). This must then also be taken into account below then.

File: reference/forms/types/options/false_values.rst.inc (hunk 3 above)
Comment (leberknecht) [reply reference/forms/types/options/false_values.rst.inc]: "doh!" again :D fixed

**Review by xabbuh — APPROVED** [review]

**Review by vudaltsov — APPROVED** [review]

**Comment by javiereguiluz** [comment]
@leberknecht thanks for this contribution! Sadly I couldn't merge it with the proprietary tool we use to merge pull requests in Symfony repos. I tried to solve it but I couldn't (for some unknown reasons, the conflict was huge with tens of conflicting files). So, I ended up replicating your changes in this other pull request: #9812. I'm really sorry but I wanted to assure you that I tried to maintain your commits and pull request but I couldn't. I'm sorry.

**Comment by leberknecht** [comment]
:D No worries!

**Pull request closed without merging**